core/src/main/scala/kafka/raft/KafkaMetadataLog.scala

```diff
@@ -173,13 +164,16 @@ final class KafkaMetadataLog private (
 
   override def truncateToLatestSnapshot(): Boolean = {
     val latestEpoch = log.latestEpoch.getOrElse(0)
-    latestSnapshotId.asScala match {
+    latestSnapshotId().asScala match {
       case Some(snapshotId) if (snapshotId.epoch > latestEpoch ||
         (snapshotId.epoch == latestEpoch && snapshotId.offset > endOffset().offset)) =>
         // Truncate the log fully if the latest snapshot is greater than the log end offset
 
         log.truncateFullyAndStartAt(snapshotId.offset)
-        oldestSnapshotId = latestSnapshotId
+        // Delete snapshot after truncating
+        val olderSnapshotIds = snapshotIds.headSet(snapshotId)
+        olderSnapshotIds.forEach(id => Snapshots.deleteSnapshotIfExists(log.dir.toPath, id))
+        snapshotIds.removeAll(olderSnapshotIds)
```

Comment:
Should we call `olderSnapshotIds.clear()` here (since the olderSnapshotIds is backed by the original snapshotIds)

Reply:
In fact, after `snapshotIds.removeAll(olderSnapshotIds)`, the  element in olderSnapshotIds is also removed.

Reply:
my point here is calling `olderSnapshotIds.clear()` instead of calling `removeAll()`

Reply:
Thank you, will fix this!